### PR TITLE
Species emote restrictions

### PR DESCRIPTION
**Pull request by ElorgRHG**
## About The Pull Request
Borrows some code from: https://github.com/ParadiseSS13/Paradise/pull/17544

Don't want teshari going around warbling at skrell now do we

## Why It's Good For The Game
image

Teshari can warble but differently due to community pushback :)

## Changelog

:cl:ElorgRHG, SabreML, lewcc
code: Certain emotes are now restricted to species.
/:cl:

### CODE REVIEW

File: code/datums/emotes.dm
Comment (francinum) [inline code/datums/emotes.dm:59]: ?

Comment (ElorgRHG) [reply code/datums/emotes.dm:59]: Forgot to change it back, oop

File: code/datums/emotes.dm
Comment (Kapu1178) [inline code/datums/emotes.dm:51]:
```suggestion
	var/list/species_type_whitelist_typecache
```

File: code/datums/emotes.dm
Comment (Kapu1178) [inline code/datums/emotes.dm:246]:
```suggestion
		if(species_type_whitelist_typecache && H.dna && !species_type_whitelist_typecache[H.dna.species])
```
is_type_in_typecache() is pointless. shitty macro. this applies to all uses of it.

**Review by Kapu1178 — CHANGES_REQUESTED** [review]

**Review by Tokorizo — APPROVED** [review]
This LGTM, ill leave it up for 24 hours to have anyone object or request more code changes.

File: code/modules/mob/emote.dm
Comment (SabreML) [inline code/modules/mob/emote.dm:51]: **[OPTIONAL]** This isn't technically related to the PR but I figured I'd suggest it anyway:

Bolding and adding a newline to these messages makes the emote list a lot easier to read, IMO:
**Before:**
before
**After:**
after
```suggestion
	var/list/message = list("<b>Available emotes, you can use them with say \"*emote\":</b> \n")
```

File: code/modules/mob/emote.dm
Comment (SabreML) [inline code/modules/mob/emote.dm:75]: **[OPTIONAL]**
Same as above:
```suggestion
		message += "\n<b><u>Species specific emotes</u></b>: \n"
```

File: code/modules/mob/living/carbon/human/emote.dm
Comment (SabreML) [inline code/modules/mob/living/carbon/human/emote.dm:97]: There seems to be a 'head tilt' emote already in `code\modules\mob\living\emote.dm`:
https://github.com/pariahstation/Pariah-Station/blob/b083e79b170e80b134e9858d274274b1eae61619/code/modules/mob/living/emote.dm#L410-L413
Maybe add these changes onto that?

File: code/modules/mob/living/carbon/human/emote.dm
Comment (SabreML) [inline code/modules/mob/living/carbon/human/emote.dm:223]: Missing the moth whitelist:
```suggestion
/datum/emote/living/chitter
	key = "chitter"
	key_third_person = "chitters"
	message = "chitters!"
	emote_type = EMOTE_AUDIBLE
	vary = TRUE
	sound = 'sound/emotes/mothchitter.ogg'
	species_type_whitelist_typecache = list(/datum/species/moth)
```

File: code/modules/mob/living/emote.dm
Comment (SabreML) [inline code/modules/mob/living/emote.dm:387]: These three are still semi-modular since they're overriding existing emotes rather than actually editing them.
The original emote definitions are here, if you could move these into them:
https://github.com/pariahstation/Pariah-Station/blob/b083e79b170e80b134e9858d274274b1eae61619/code/modules/mob/living/emote.dm#L358-L362
https://github.com/pariahstation/Pariah-Station/blob/b083e79b170e80b134e9858d274274b1eae61619/code/modules/mob/living/emote.dm#L364-L370
https://github.com/pariahstation/Pariah-Station/blob/b083e79b170e80b134e9858d274274b1eae61619/code/modules/mob/living/emote.dm#L44-L48

File: code/modules/mob/living/emote.dm
Comment (SabreML) [inline code/modules/mob/living/emote.dm:666]: Another one here:
https://github.com/pariahstation/Pariah-Station/blob/b083e79b170e80b134e9858d274274b1eae61619/code/modules/mob/living/emote.dm#L347-L351

File: code/modules/mob/living/emote.dm
Comment (ElorgRHG) [reply code/modules/mob/living/emote.dm]: I love modularisation

File: code/modules/mob/living/emote.dm
Comment (SabreML) [inline code/modules/mob/living/emote.dm:58]: No need to return at the end
```suggestion
/datum/emote/living/burp/get_sound(mob/living/user)
	if(iscarbon(user))
		if(user.gender == MALE)
			return 'sound/emotes/male/burp_m.ogg'
		return 'sound/emotes/female/burp_f.ogg'

/datum/emote/living/choke
```

File: code/modules/mob/living/emote.dm
Comment (SabreML) [inline code/modules/mob/living/emote.dm:211]:
```suggestion
/datum/emote/living/gasp/get_sound(mob/living/user)
	if(iscarbon(user))
		if(user.gender == MALE)
			return pick('sound/emotes/male/gasp_m1.ogg',
						'sound/emotes/male/gasp_m2.ogg',
						'sound/emotes/male/gasp_m3.ogg',
						'sound/emotes/male/gasp_m4.ogg',
						'sound/emotes/male/gasp_m5.ogg',
						'sound/emotes/male/gasp_m6.ogg')

		return pick('sound/emotes/female/gasp_f1.ogg',
					'sound/emotes/female/gasp_f2.ogg',
					'sound/emotes/female/gasp_f3.ogg',
					'sound/emotes/female/gasp_f4.ogg',
					'sound/emotes/female/gasp_f5.ogg',
					'sound/emotes/female/gasp_f6.ogg')

/datum/emote/living/giggle
```

File: code/modules/mob/living/emote.dm
Comment (SabreML) [inline code/modules/mob/living/emote.dm:372]:
```suggestion
/datum/emote/living/sigh/get_sound(mob/living/user)
	if(iscarbon(user))
		if(user.gender == MALE)
			return 'sound/emotes/male/male_sigh.ogg'
		return 'sound/emotes/female/female_sigh.ogg'

/datum/emote/living/clap1
```

File: code/modules/mob/living/emote.dm
Comment (SabreML) [inline code/modules/mob/living/emote.dm:426]:
```suggestion
/datum/emote/living/sniff/get_sound(mob/living/user)
	if(iscarbon(user))
		if(user.gender == MALE)
			return 'sound/emotes/male/male_sniff.ogg'
		return 'sound/emotes/female/female_sniff.ogg'

/datum/emote/living/snore
```

**Review by SabreML — COMMENTED** [review]
99% just styling nitpicks now, it should be able to be merged afterwards.

File: code/modules/mob/living/carbon/human/emote.dm
Comment (ElorgRHG) [reply code/modules/mob/living/carbon/human/emote.dm]: I blame this on the existance of teshari

File: code/modules/mob/living/emote.dm
Comment (ElorgRHG) [reply code/modules/mob/living/emote.dm]: I shall comment out code (:trollface:) for use when Vox get added

File: code/modules/mob/living/carbon/human/emote.dm
Comment (Kapu1178) [inline code/modules/mob/living/carbon/human/emote.dm:247]: this isn't a typecache, this is a list.

Comment (SabreML) [reply code/modules/mob/living/carbon/human/emote.dm:247]: `mob_type_allowed_typecache`, `mob_type_blacklist_typecache`, and `mob_type_ignore_stat_typecache` all get made into typecaches in `/datum/emote/New()`:
https://github.com/pariahstation/Pariah-Station/blob/434e9eecaf65c0b0b0091b32585ff2054db7811d/code/datums/emotes.dm#L65-L75

File: code/modules/mob/living/carbon/human/emote.dm
Comment (miniusAreas) [inline code/modules/mob/living/carbon/human/emote.dm:261]: Share with teshari. Share concerns against with Teshari lore maintainer.

File: code/modules/mob/living/carbon/human/emote.dm
Comment (miniusAreas) [inline code/modules/mob/living/carbon/human/emote.dm:270]: Share with Teshari. Share concerns against with Teshari lore maintainer.

File: code/modules/mob/living/carbon/human/emote.dm
Comment (miniusAreas) [inline code/modules/mob/living/carbon/human/emote.dm:279]: Share with Teshari. Share concerns against with Teshari lore maintainer.

**Review by miniusAreas — CHANGES_REQUESTED** [review]
Go through these, and have them shared with Teshari. If you have any concerns / oppositions to this, please bring them up with me on Discord. In the future, please consult the various lore maintainers for species before making a change that can effect roleplay. Thank you.

File: code/modules/mob/living/carbon/human/emote.dm
Comment (ElorgRHG) [reply code/modules/mob/living/carbon/human/emote.dm:261]: I have no means to restrict it to teshari considering https://github.com/pariahstation/Pariah-Station/pull/688 is not yet merged.

File: code/modules/mob/living/carbon/human/emote.dm
Comment (ElorgRHG) [reply code/modules/mob/living/carbon/human/emote.dm:270]: Same as above

**Review by francinum — CHANGES_REQUESTED** [review]
Enforcement of Minius's change request. This can wait until tesh are merged, though. Which should hopefully not be much longer.

File: code/modules/mob/living/emote.dm
Comment (SabreML) [inline code/modules/mob/living/emote.dm:195]: A lot of these could really do with having early returns, but considering how many there are that's probably out of scope of this PR.

File: code/modules/mob/living/emote.dm
Comment (SabreML) [inline code/modules/mob/living/emote.dm:289]: Having just said that of course, this one is checking `H.mind` and `H.mind.miming` twice in a row, so maybe it'd be worth changing *one* of them:
```dm
/datum/emote/living/laugh/get_sound(mob/living/user)
	if(!ishuman(user) || user.mind?.miming)
		return null

	var/mob/living/carbon/human/H = user
	if(H.dna.species.id == SPECIES_HUMAN)
		if(user.gender == FEMALE)
			return 'sound/voice/human/womanlaugh.ogg'
		else
			return pick('sound/voice/human/manlaugh1.ogg', 'sound/voice/human/manlaugh2.ogg')
	if(H.dna.species.id == SPECIES_MOTH)
		return 'sound/emotes/mothlaugh.ogg'
```

**Review by SabreML — COMMENTED** [review]
Last nitpicks I swear

**Review by SabreML — APPROVED** [review]

**Pull request merged**
